lib/usd/ui/layerEditor/mayaLayerEditorWindow.cpp

```diff
@@ -118,6 +118,7 @@ void MayaLayerEditorWindow::onClearUIOnSceneReset()
     LayerTreeModel::suspendUsdNotices(true);
     setCentralWidget(nullptr);
     delete _layerEditor;
+    _sessionState.setStageEntry(UsdLayerEditor::SessionState::StageEntry{});
```

Comment:
The stage entry was holding a ref on the stage, which prevented it from being fully released on File->Open. If re-opening the same file or a file referencing the same stage, the leak caused any unsaved modifications to outlive the File->Open action.